controllers/util/solr_util.go

```diff
@@ -134,7 +135,7 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 
 	// Keep track of the SolrOpts that the Solr Operator needs to set
 	// These will be added to the SolrOpts given by the user.
-	allSolrOpts := []string{"-DhostPort=$(SOLR_NODE_PORT)"}
```

Comment:
So we can actually just use an envVar, `SOLR_PORT_ADVERTISE`, instead of relying on SOLR_OPTS. Makes the code a small bit cleaner 🙂 

But we can keep the `hostPort` in place and deprecate it (So we would be using `-DhostPort` and `SOLR_PORT_ADVERTISE` at the same time), to give people with a custom `solr.xml` a release to change their Solr.xml to use `solr.port.advertise`, like is used in the official `solr.xml`. As long as we document it in the upgrade notes, we should be good to go!

Reply:
Ah, right!  I forgot `bin/solr` was set up to convert `SOLR_PORT_ADVERTISE` to a sysprop that solr.xml can take advantage of, cool.  👍